src/commands/workflow/new.ts

```diff
@@ -1,99 +1,300 @@
+        initial: slugify(name),
+        validate: (value: string) => {
+          if (!value || value.trim().length === 0) {
+            return "Workflow key is required";
+          }
+
+          const keyError = Workflow.validateWorkflowKey(value);
+          if (keyError) {
+            return `Invalid workflow key: ${keyError}`;
+          }
+
+          return true;
+        },
+      });
+      key = keyResponse.key;
+    }
+
+    // Validate the workflow key
+    const workflowKeyError = Workflow.validateWorkflowKey(key);
     if (workflowKeyError) {
       return this.error(
-        `Invalid workflow key \`${args.workflowKey}\` (${workflowKeyError})`,
+        `Invalid workflow key \`${key}\` (${workflowKeyError})`,
+      );
+    }
+
+    const workflowDirCtx = await this.getWorkflowDirContext(key);
+
+    // Check if the workflow directory already exists, and prompt to confirm if not.
+    if (workflowDirCtx.exists) {
+      this.log(
+        `‣ Found \`${workflowDirCtx.key}\` at ${workflowDirCtx.abspath}`,
       );
+    } else {
+      const prompt = `Create a new workflow directory \`${workflowDirCtx.key}\` at ${workflowDirCtx.abspath}?`;
+      const input = flags.force || (await promptToConfirm(prompt));
+      if (!input) return;
     }
 
-    // 3. Parse and validate the steps flag, if given.
-    const [steps, stepsError] = Workflow.parseStepsInput(flags.steps || "");
-    if (stepsError) {
-      return this.error(`Invalid --steps \`${flags.steps}\` (${stepsError})`);
+    // Generate the workflow either from a template or from scratch
+    await (flags.template
+      ? this.fromTemplate(workflowDirCtx, name, flags.template)
+      : this.fromEmpty(workflowDirCtx, name));
+
+    if (flags.push) {
+      spinner.start("‣ Pushing workflow to Knock");
+
+      try {
+        await WorkflowPush.run([key]);
+      } catch (error) {
+        this.error(`Failed to push workflow to Knock: ${error}`);
+      } finally {
+        spinner.stop();
+      }
     }
 
-    // 4. Ensure not to overwrite any existing path accidentally.
-    const newWorkflowDirPath = path.resolve(cwd, args.workflowKey);
-    const pathExists = await fs.pathExists(newWorkflowDirPath);
-    if (pathExists && !flags.force) {
-      return this.error(
-        `Cannot overwrite an existing path at ${newWorkflowDirPath}` +
-          " (use --force to overwrite)",
+    this.log(`‣ Successfully created workflow \`${key}\``);
+  }
+
+  async fromTemplate(
+    workflowDirCtx: WorkflowDirContext,
+    name: string,
+    templateString: string,
+  ): Promise<void> {
+    // When being called from the template string, we want to try and generate
+    // the workflow from the provided template.
+    const channelsByType = await this.listAllChannelsByType();
+
+    spinner.start(`‣ Generating workflow from template \`${templateString}\``);
+
+    try {
+      await Workflow.generateWorkflowFromTemplate(
+        workflowDirCtx,
+        templateString,
```

Comment:
We're passing in `templateString` transparently here, so we technically don't know if this template is a workflow template. I see that in [our repo](https://github.com/knocklabs/templates) we only have templates for workflows, so not necessarily a huge issue. I'm guessing we will want to support templates for other resources, so we will probably want to have a validation step to ensure we have the right type of template string but we can worry about that on another day.